stories/spectra/mass.stories.tsx

```diff
@@ -43,7 +45,35 @@ export function AdvancedMassExample() {
         yAxis="y"
         lineStyle={{ stroke: 'red' }}
       />
-      <Axis id="x" position="bottom" label="Mass [m/z]" />
+      <Annotations>
+        {advancedData.bestPeaks.map((peak) => (
+          <>
+            <Line
+              key={peak.label}
+              x1={peak.x}
+              x2={peak.x}
+              y1={peak.y}
+              y2={peak.y + 2}
```

Comment:
If you put number a properties for x1, x2, y1 and y2 the shift will be based on the values of the axis. This means that in your case if you zoom the labels will be exploded.

This is why we can put either number of strings for the x1, x2, y1, y2 properties. If it is strings the values are in pixels from the (0,0) top left.

This means that we need to create the labels containing:
- the line
- the first text
- the second text
using the "pixels" (a string) to shift the different annotations

And they should be placed in a Groupd and then placed using the value of the axis (meaining x and y are numbers)

https://github.com/zakodium/react-plot/blob/main/src/components/Annotations/Group.tsx

Reply:
fixed